node/builder_chain.go

```diff
@@ -285,6 +286,9 @@ func ConfigFullNode(c interface{}) Option {
 		If(cfg.FaultReporter.EnableConsensusFaultReporter,
 			Override(ConsensusReporterKey, modules.RunConsensusFaultReporter(cfg.FaultReporter)),
 		),
+
+		Override(new(chainindex.Indexer), modules.ChainIndexer(cfg.Index)),
+		Override(InitChainIndexerKey, modules.InitChainIndexer),
 	)
```

Comment:
they should probably be in the `isFullNode` guard, and that's probably true for the existing `MsgIndex` but at least that's a default-off thing that a lite node probably would never enable.